projects/python/perception/object_tracking_3d/demos/vpit_single_object_tracking/demo.py

```diff
@@ -0,0 +1,581 @@
+
+    return wrapped
+
+
+def draw_fps(frame, fps):
+    cv2.putText(
+        frame,
+        f"{fps:.1f} FPS",
+        (10, frame.shape[0] - 10),
+        cv2.FONT_HERSHEY_SIMPLEX,
+        1,
+        TEXT_COLOR,
+        1,
+    )
+
+
+def draw_dict(frame, dict, scale=5):
+
+    i = 0
+
+    for k, v in dict.items():
+        cv2.putText(
+            frame,
+            f"{k}: {v}",
+            (10, frame.shape[0] - 10 - 30 * scale * i),
+            cv2.FONT_HERSHEY_SIMPLEX,
+            scale,
+            TEXT_COLOR,
+            scale,
+        )
+        i += 1
+
+
+def stack_images(images, mode="horizontal"):
+
+    max_width, max_height = 0, 0
+
+    for image in images:
+        width, height, _ = image.shape
+        max_width = max(max_width, width)
+        max_height = max(max_height, height)
+
+    if mode == "horizontal":
+        for i in range(len(images)):
+            width, _, _ = images[i].shape
+
+            delta = max_width - width
+            pad = delta // 2
+
+            images[i] = np.pad(
+                images[i], [(pad, pad + delta % 2), (0, 0), (0, 0)]
+            )
+
+        return cv2.hconcat(images)
+    elif mode == "vertical":
+        for i in range(len(images)):
+            _, height, _ = images[i].shape
+
+            delta = max_height - height
+            pad = delta // 2
+
+            images[i] = np.pad(
+                images[i], [(0, 0), (pad, pad + delta % 2), (0, 0)]
+            )
+
+        return cv2.vconcat(images)
+
+
+def vpit_single_object_detection_3d(params="best", model_name=None):
+    global point_cloud_generator, output_frame, lock, lidar_type
+
+    with lock:
+        output_frame = np.zeros((400, 400, 3), dtype=np.uint8)
+        draw_dict(output_frame, {"Loading": "model"}, 1)
+
+    # Prep stats
+    fps = runnig_fps()
+
+    cv2.namedWindow("VPIT", cv2.WINDOW_NORMAL)
+
+    predict = model_name is not None and model_name != "None"
+
+    if predict:
+
+        if params == "best":
+            backbone = "pp"
+            device = "cuda:0" if torch.cuda.is_available() else "cpu"
+            tracking_learner = ObjectTracking3DVpitLearner(
+                model_config_path=backbone_configs[backbone],
+                device=device,
+                backbone=backbone,
+                extrapolation_mode="linear+",
+                window_influence=0.85,
+                score_upscale=8,
+                rotation_penalty=0.98,
+                target_feature_merge_scale=0.0,
+                min_top_score=None,
+                offset_interpolation=0.25,
+                search_type="small",
+                target_type="normal",
+                feature_blocks=1,
+                target_size=[-1, -1],
+                search_size=[-1, -1],
+                context_amount=0.25,
+                lr=0.0001,
+                r_pos=2,
+                augment=None,
+            )
+        else:
+            tracking_learner = ObjectTracking3DVpitLearner()
```

Comment:
This will probably fail if the else code is ever run, it's missing the `model_config_path` argument.